tests/test_riskpool_lifecycle.py

```diff
@@ -293,6 +293,367 @@ def test_suspend_resume(
+    
+
+def test_pause_archive_as_owner(
+    instance: GifInstance, 
+    testCoin,
+    gifTestProduct: GifTestProduct, 
+    productOwner,
+    riskpoolKeeper: Account,
+    owner: Account,
+    customer: Account,
+    feeOwner: Account,
+    capitalOwner: Account
+):
+    riskpool = gifTestProduct.getRiskpool().getContract()
+    riskpoolId = riskpool.getId()
+
+    instanceService = instance.getInstanceService()
+    instanceOperatorService = instance.getInstanceOperatorService()
+    assert instanceService.getComponentState(riskpoolId) == 3
+
+    initialFunding = 10000
+    bundleOwner = riskpoolKeeper
+    fund_riskpool(instance, owner, capitalOwner, riskpool, bundleOwner, testCoin, initialFunding)
+
+    riskpool.bundles() == 1
+    bundle = riskpool.getBundle(0)
+    print(bundle)
+
+    (
+        bundleId,
+        riskpoolId,
+        tokenId,
+        state,
+        filter,
+        capital,
+        lockedCapital,
+        balance,
+        createdAt,
+        updatedAt
+    ) = bundle
+
+    # check bundle values with expectation
+    assert bundleId == 1
+    assert riskpoolId == riskpool.getId()
+
+    # verify that attributes directly from riskpool match with 
+    # bundle attributes from instance service
+    bundleFromCore = instanceService.getBundle(bundleId).dict()
+    print(bundleFromCore)
+
+    assert bundleFromCore["id"] == bundleId
+    assert bundleFromCore["riskpoolId"] == riskpoolId
+    assert bundleFromCore["tokenId"] == tokenId
+    assert bundleFromCore["state"] == state
+    assert bundleFromCore["filter"] == filter
+    assert bundleFromCore["capital"] == capital
+    assert bundleFromCore["lockedCapital"] == lockedCapital
+    assert bundleFromCore["balance"] == balance
+    assert bundleFromCore["createdAt"] == createdAt
+    assert bundleFromCore["updatedAt"] == updatedAt
+
+    componentOwnerService = instance.getComponentOwnerService()
+
+    # create test policy before riskpool is paused
+    product = gifTestProduct.getContract()
+    policyId = apply_for_policy(instance, owner, product, customer, testCoin, 100, 1000)
+
+    # ensure that owner may not archive riskpool
+    assert owner != riskpoolKeeper
+    with brownie.reverts("ERROR:CMP-015:ACTIVE_INVALID_TRANSITION"):
+        instanceOperatorService.archive(riskpoolId, {'from':owner})
+
+    with brownie.reverts("ERROR:CMP-015:ACTIVE_INVALID_TRANSITION"):
+        componentOwnerService.archive(riskpoolId, {'from':riskpoolKeeper})
+
+    assert instanceService.getComponentState(riskpoolId) == 3
+
+    # ensure that riskpool keeper may pause riskpool
+    componentOwnerService.pause(riskpoolId, {'from':riskpoolKeeper})
+    assert instanceService.getComponentState(riskpoolId) == 4
+
+    componentOwnerService.archive(riskpoolId, {'from':riskpoolKeeper})
+    assert instanceService.getComponentState(riskpoolId) == 6
+
+    # ensure that riskpool actions are blocked for paused riskpool

```

Comment:
maybe need to change implementation: only allow archiving when all bundles are burned?